modules/product/src/main/java/com/opengamma/strata/product/bond/CapitalIndexedBondTrade.java

```diff
@@ -0,0 +1,515 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.bond;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableDefaults;
+import org.joda.beans.ImmutableValidator;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.market.ReferenceData;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.id.LinkResolver;
+import com.opengamma.strata.collect.id.StandardId;
+import com.opengamma.strata.product.ResolvableTrade;
+import com.opengamma.strata.product.SecurityLink;
+import com.opengamma.strata.product.SecurityTrade;
+import com.opengamma.strata.product.TradeInfo;
+import com.opengamma.strata.product.rate.RateObservation;
+
+/**
+ * A trade representing a capital indexed bond.
+ * <p>
+ * A trade in an underlying {@link CapitalIndexedBond}.
+ */
+@BeanDefinition
+public final class CapitalIndexedBondTrade
+    implements SecurityTrade<CapitalIndexedBond>, ResolvableTrade<ResolvedCapitalIndexedBondTrade>, ImmutableBean, Serializable {
+
+  /**
+   * The additional trade information, defaulted to an empty instance.
+   * <p>
+   * This allows additional information to be attached to the trade.
+   */
+  @PropertyDefinition(overrideGet = true)
+  private final TradeInfo tradeInfo;
+  /**
+   * The link to the capital indexed bond that was traded.
+   * <p>
+   * This property returns a link to the security via a {@link StandardId}.
+   * See {@link #getSecurity()} and {@link SecurityLink} for more details.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final SecurityLink<CapitalIndexedBond> securityLink;
+  /**
+   * The quantity, indicating the number of bond contracts in the trade.
+   * <p>
+   * This will be positive if buying and negative if selling.
+   */
+  @PropertyDefinition
+  private final long quantity;
+
```

Comment:
Do we need a "payment" of some type to represent the settlement (price, date, etc.)?